src/astarte_bson_serializer.c

```diff
@@ -157,17 +162,17 @@ void astarte_bson_serializer_append_end_of_document(struct astarte_bson_serializ
 {
     astarte_byte_array_append_byte(&bs->ba, '\0');
 
-    uint8_t *sizeBuf = NULL;
-    INT32_TO_BYTES(bs->ba.size, sizeBuf)
+    uint8_t sizeBuf[4];
+    uint32_to_bytes(bs->ba.size, sizeBuf);
```

Comment:
Is it right use `uint32_to_bytes` instead of `int32_to_bytes`?

Reply:
It makes more sense to use an unsigned number to store a size as it should never be negative.